### PR TITLE
Enable passing the reportedUuid from cattle

### PR DESCRIPTION
**Pull request by pranavs18**
PR for enabling JWT-auth by sending the reportedUuid

### CODE REVIEW

File: code/implementation/host-api/src/main/java/io/cattle/platform/host/service/impl/HostApiServiceImpl.java [hunk 1]
```diff
@@ -66,7 +68,13 @@ public HostApiAccess getAccess(Long hostId, Map<String,Object> data) {
...
     protected String getToken(Host host, Map<String,Object> inputData) {
         Map<String,Object> data = new HashMap<String,Object>(inputData);
-        data.put(HOST_UUID, host.getUuid());
+        String uuid = DataAccessor.fields(host).withKey(HostConstants.FIELD_REPORTED_UUID).as(String.class);
+        if(uuid != null){
+            data.put(HostConstants.FIELD_REPORTED_UUID,uuid);
+        }
+        else{
+            data.put(HOST_UUID, host.getUuid());
+        }
```
Comment (ibuildthecloud) [inline code/implementation/host-api/src/main/java/io/cattle/platform/host/service/impl/HostApiServiceImpl.java]: The key should always be HOST_UUID, the only that should be conditional is from where the uuid is obtained.

Comment (pranavs18) [inline code/implementation/host-api/src/main/java/io/cattle/platform/host/service/impl/HostApiServiceImpl.java]: Oh, i see. This is the reason, I had to put that check on ReportedUuid and HostUuid in GO code.I'll change this to HOST_UUID and modify the GO code accordingly.

**Comment by pranavs18** [comment]
Updated the PR as per the comments. Drone is failing on test_docker_volume, need to check that.

**Comment by pranavs18** [comment]
Rebased Cattle and drone passes. @ibuildthecloud !

**Pull request merged**